Dart/src/com/jetbrains/lang/dart/ide/devtools/DevToolsService.kt

```diff
@@ -0,0 +1,87 @@
+// Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+package com.jetbrains.lang.dart.ide.devtools
+
+import com.google.gson.JsonObject
+import com.google.gson.JsonParser
+import com.intellij.execution.configurations.GeneralCommandLine
+import com.intellij.execution.process.ColoredProcessHandler
+import com.intellij.execution.process.ProcessEvent
+import com.intellij.execution.process.ProcessListener
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Key
+import com.intellij.openapi.util.io.FileUtil
+import com.intellij.openapi.util.text.StringUtil
+import com.jetbrains.lang.dart.sdk.DartSdk
+import io.flutter.run.daemon.DevToolsInstance
+import java.nio.charset.StandardCharsets
+import java.util.concurrent.CompletableFuture
+
+@Service(Service.Level.PROJECT)
+class DevToolsService(private val myProject: Project) : Disposable {
+  private val MIN_SDK_VERSION: String = "2.15"
+  private lateinit var processHandler: ColoredProcessHandler;
+
+  private val devToolsFuture: CompletableFuture<DevToolsInstance> = CompletableFuture()
+
+  fun startService(dtdUri: String? = null): Void? {
+    val sdk = DartSdk.getDartSdk(myProject)
+    if (sdk == null || !isDartSdkVersionSufficient(sdk)) {
+      return null
+    }
+    val result = GeneralCommandLine().withWorkDirectory(sdk.homePath)
+    result.charset = StandardCharsets.UTF_8
+    result.exePath = FileUtil.toSystemDependentName("dart")
+    result.addParameter("devtools")
+    result.addParameter("--machine")
+    if (dtdUri != null) {
+      result.addParameter("--dtd-uri=$dtdUri")
+    }
+
+    processHandler = ColoredProcessHandler(result)
+    processHandler.addProcessListener(object : ProcessListener {
+      override fun onTextAvailable(event: ProcessEvent, outputType: Key<*>) {
+        val text = event.text.trim { it <= ' ' }
```

Comment:
Is this comment valid for this service as well?
```
// The first line of text is the command issued, which can be ignored.
```

Reply:
Yes, also valid for the DevTools service.

Reply:
Ok, I'll add some `if` tomorrow, like in DTDService

Reply:
done